_posts/2023-06-20-resume-parsing-insights-and-steps-to-create-your-own-parser.md

```diff
@@ -0,0 +1,896 @@
+The extracted data is then converted into a structured format, allowing for easy analysis and integration into recruitment systems.
+
+## Benefits of Resume Parsing
+
+- It is a time-saving automation
+- It increases efficiency in candidate screening
+- Improves accuracy in data extraction
+- It standardizes the data extraction and formatting
+
+
+## What you'll learn from this blog:
+
+1. Resume parsing techniques for different file formats.
+2. Extracting specific details from resumes.
+3. Leveraging NLP techniques for parsing.
+4. Handling multicolumn resumes.
+5. Dockerizing the Application: Simplifying Deployment and Scalability
+6. Hosting it on AWS EC2.
+
+**Let's get Started 🎉**
+
+We'll utilize Python and its Flask framework to create a resume parsing server.
+
+## Application Flow Chart:
+
+![Application Flow Chart Image]( {{site.baseurl}}/images/resume-parsing-insights-and-steps-to-create-your-own-parser/file-flow.jpg)
+
+We will be primarily working on 3 categories of file formats:
+1. PDF
+2. DOCX
+3. Images (.png, .jpg, etc.)
+
+### Data that we will be extracting
+
+1. Embedded links in PDF
+2. Personal data: <br>
+    2.1. Name: First name and last name <br>
+    2.2. Email <br>
+    2.3. Phone Number <br>
+    2.4. Address: City, Country, and Zip code <br>
+    2.5. Links: Social and Coding Platform links <br>
+3. Education <br>
+    3.1. Institute name <br>
+    3.2. Duration: Start date and End date <br>
+    3.3. Grade/CGPA <br>
+    3.4. Degree <br>
+4. Experience<br>
+    4.1. Company name<br>
+    4.2. Role<br>
+    4.3. Durations: Start date and End date<br>
+    4.4. Skills<br>
+5. Certification: <br>
+    5.1. Description <br>
+    5.2. Duration <br>
+    5.3. Skill <br>
+6. Project: <br>
+    6.1. Project name <br>
+    6.2. Skills <br>
+    6.3. Description <br>
+7. Skills
+8. Achievements
+9. Exam scores<br>
+    9.1. Exam name<br>
+    9.2 Score<br>
+10. All other sections present in resume 
+
+## Date/Duration Extraction
+
+To extract dates from text, we will use `datefinder` module, and regexp to extract years.
+Then we will combine these two and sort dates to get start and end date for our duration.
+
+```python
+import re
+from datetime import date
+import datefinder
+
+
+def get_date(input_string):
+    '''Get date from text'''
+    matches = list(datefinder.find_dates(input_string))
+
+    res = []
+    for i in matches:
+        date_str = str(i).split(' ')
+        extracted_date = date_str[0]
+
+        res.append(extracted_date)
+    return res
+
+
+def get_years(txt):
+    '''Get years from text'''
+    pattern = r'[0-9]{4}'
+    lst = re.findall(pattern, txt)
+
+    current_date = date.today()
+    current_year = current_date.year
+    res = []
+    for i in lst:
+        year = int(i)
+        if 1900 <= year <= (current_year + 10):
+            res.append(i + "-01-01")
```

Comment:
This hard coded line here produce undesirable results.

Even if some resume have the month along with year. Resume parser override the month and date with 01-01

Reply:
The above piece of code is to only detect year i.e. in case complete date is not given, if complete date has been provided, similar to the example that you have taken, it will be captured by the `get_date` function on line `95`.